virttest/utils_net.py

```diff
@@ -3744,3 +3746,190 @@ def map_hostname_ipaddress(hostname_ip_dict, session=None):
+
+        search_res = re.findall(search_exp, line)
+        if not search_res:
+            continue
+        try:
+            value_dict = dict(map(_convert_fun, search_res))
+            MSIs_number = value_dict["MSIs"]
+            queues_number = value_dict["queues"]
+            return (MSIs_number, queues_number)
+        except Exception:
+            pass
+    return (None, None)
+
+
+def get_MSIs_and_queues_windows(test, params, vm, timeout=360):
+    """
+    Get msi&queues infomation of currentwindows guest.
+    First start a traceview session, then restart the nic interface
+    to trigger logging. By analyzing the dumped output, the msi& queue
+    info is acquired.
+
+    :param test: the runing test
+    :param params: the test params
+    :param vm: target vm
+    :param timeout: the timeout of login
+    :return: a tuple of (msis, queues)
+    """
+
+    session = vm.wait_for_login(timeout=timeout)
+
+    # prepare traceview environment
+    _prepare_traceview_windows(test, params, vm, session, timeout)
+    session.close()
+
+    # start traceview
+    logging.info("Start trace view with pdb file")
+    session_serial = vm.wait_for_serial_login(timeout=timeout)
+    log_path = "c:\\logfile.etl"
+    clean_cmd = "del %s" % log_path
+    session_serial.cmd(clean_cmd, timeout=timeout)
+    traceview_local_path = "c:\\traceview.exe"
+    pdb_local_path = "c:\\netkvm.pdb"
+    start_traceview_cmd = "%s -start test_session -pdb %s -level 5 -flag 0x1fff -f %s"
+    start_traceview_cmd %= (traceview_local_path, pdb_local_path, log_path)
+    session_serial.cmd(start_traceview_cmd, timeout=timeout)
+
+    # restart nic
+    logging.info("Restart guest nic")
+    mac = vm.get_mac_address(0)
+    connection_id = get_windows_nic_attribute(session_serial,
+                                              "macaddress",
+                                              mac,
+                                              "netconnectionid")
+    restart_windows_guest_network(session_serial, connection_id)
+
+    # stop traceview
+    logging.info("Stop traceview")
+    stop_traceview_cmd = "%s -stop test_session" % traceview_local_path
+    session_serial.cmd(stop_traceview_cmd, timeout=timeout)
+
+    # checkout traceview output
+    logging.info("Check etl file generated by traceview")
+    dump_file = "c:\\trace.txt"
+    _remove_file(session_serial, dump_file)
+    dump_cmd = "%s -process %s -pdb %s -o %s"
+    dump_cmd %= (traceview_local_path, log_path, pdb_local_path, dump_file)
+    status, output = session_serial.cmd_status_output(dump_cmd)
+    if status:
+        logging.error("Cann't dump log file %s: %s" % (log_path, output))
+    time.sleep(100)
+    session_serial.cmd("taskkill /im traceview.exe")
+    time.sleep(10)
+    status, output = session_serial.cmd_status_output("type %s" % dump_file)
+    if status:
+        logging.error("Cann't read dumped file %s: %s" % (dump_file, output))
```

Comment:
Same for these `logging.error`s, we need raise exception or return False/None to interrupt the execution when it can't continue the following steps.